test/test_fx.py

```diff
@@ -2529,15 +2529,15 @@ def test_profiler_ranges_side_effect(self):
             if node.op == 'call_function':
                 found_targets.setdefault(node.target)
         self.assertEqual(
-            found_targets.keys(), [torch.ops.profiler._record_function_enter, torch.ops.profiler._record_function_exit])
+            found_targets.keys(), {torch.ops.profiler._record_function_enter, torch.ops.profiler._record_function_exit})
```

Comment:
Not sure if I'm missing anything here, but comparing the keys of a dictionary, which are unordered by definition, to an ordered sequence, is a recipe for a flaky test. @jamesr66a do we actually want the test the order here?

Reply:
@pmeier PyTorch requires at minimum Python 3.6. Dictionaries are insertion-ordered by implementation in CPython 3.6 (https://stackoverflow.com/q/39980323) and this is codified as language standard in 3.7+ (https://mail.python.org/pipermail/python-dev/2017-December/151283.html). Thus, there is no risk of flakiness here.

Reply:
Thanks for the info, I didn't know that. Still, it makes a difference for the testing: the old implementation of `assertEqual` checked for iterables, but treated them as sequences. The new implementation will no longer do that. Meaning, both inputs either have to be set-like if the order doesn't matter or sequences if it does:

```python
>>> d = dict(foo="bar", baz=0)
>>> d.keys() == ["foo", "baz"]
False
>>> d.keys() == {"foo", "baz"}
True
>>> list(d.keys()) == ["foo", "baz"]
True
```